modules/tinymce/src/core/main/ts/content/ContentTypes.ts

```diff
@@ -10,23 +10,42 @@ import AstNode from '../api/html/Node';
 export type Content = string | AstNode;
 export type ContentFormat = 'raw' | 'text' | 'html' | 'tree';
 
-export interface GetContentArgs {
-  format?: ContentFormat;
-  get?: boolean;
-  content?: string;
-  getInner?: boolean;
+interface GetContentArgsBase {
+  format: ContentFormat;
+  get: boolean;
+  getInner: boolean;
   no_events?: boolean;
   [key: string]: any;
 }
 
-export interface SetContentArgs {
-  format?: string;
-  set?: boolean;
-  content?: string;
+interface SetContentArgsBase {
+  format: string;
```

Comment:
I know that by changing everything to `Partial` you've managed to avoid any breakages to actual editor APIs — but I'm wondering what the semver implications are of changing the type itself even if the APIs stay the same. Should we find a way to mention that all the fields on these types are now required (while making it clear that you can pass `Partial` types into all of our editor APIs) in the changelog?

Reply:
@ashfordneil also remember this is for 6.0 and we are looking to split the tree content out into a separate API, so a large amount of this is temporary and why it's not getting merged to a main branch.